engine/src/main/java/org/hibernate/validator/internal/metadata/BeanMetaDataManagerImpl.java

```diff
@@ -146,7 +146,14 @@ public BeanMetaDataManagerImpl(ConstraintCreationContext constraintCreationConte
 	public <T> BeanMetaData<T> getBeanMetaData(Class<T> beanClass) {
 		Contracts.assertNotNull( beanClass, MESSAGES.beanTypeCannotBeNull() );
 
-		BeanMetaData<T> beanMetaData = (BeanMetaData<T>) beanMetaDataCache.computeIfAbsent( beanClass,
+		// First, let's do a simple lookup as it's the default case
+		BeanMetaData<T> beanMetaData = (BeanMetaData<T>) beanMetaDataCache.get( beanClass );
+
+		if ( beanMetaData != null ) {
+			return beanMetaData;
+		}
+
+		beanMetaData = (BeanMetaData<T>) beanMetaDataCache.computeIfAbsent( beanClass,
 				bc -> createBeanMetaData( bc ) );
```

Comment:
This one is interesting: this code creates a new instance of the lambda for each call so we end up creating a lot of lambda instances.

Reply:
Right. So I switched to that but I get the previous value from `putIfAbsent` and return that if there is one.